cmd/kaidump/kaidump.go

```diff
@@ -1,75 +1,93 @@
-	flag.Parse()
-
-	// Initialize and write to Vault stores for every component
-	initConfig()
-	secretStore := getVaultService("global")
-	apps, err := secretStore.ListAppNames()
-	if err != nil {
-		panic(err)
-	}
-	scopes := os.Getenv("KAIGARA_SCOPES")
-	if scopes == "" {
-		panic("Undefined KAIGARA_SCOPES env")
-	}
-	scopesList := strings.Split(scopes, ",")
-	if len(scopesList) <= 0 {
-		panic("Wrong KAIGARA_SCOPES")
-	}
-	appMap := make(map[string]interface{})
-	for _, app := range apps {
-		scopeMap := make(map[string]interface{})
-		for _, scope := range scopesList {
-			err := secretStore.LoadSecrets(app, scope)
-			if err != nil {
-				panic(err)
-			}		
-			secrets, err := secretStore.GetSecrets(app, scope)
-			if err != nil {
-				panic(err)
-			}
-			scopeMap[scope] = secrets
-		}
-		appMap[app] = scopeMap
-	}
-	res, err := yaml.Marshal(&appMap)
-	if err != nil {
-		panic(err)
-	}
-	fmt.Printf("--- secrets dump:\n%s\n\n", string(res))
-	err = ioutil.WriteFile(*filepath, res, 0644)
-	if err != nil {
-		panic(err)
-	}
-}
+package main
+
+import (
+	"flag"
+	"fmt"
+	"io/ioutil"
+
+	"github.com/openware/kaigara/pkg/config"
+	"github.com/openware/kaigara/pkg/vault"
+
+	"github.com/openware/pkg/ika"
+	"strings"
+	"gopkg.in/yaml.v3"
+)
+
+var cnf = &config.KaigaraConfig{}
+
+func initConfig() {
+	err := ika.ReadConfig("", cnf)
+	if err != nil {
+		panic(err)
+	}
+}
+
+func getVaultService(appName string) *vault.Service {
+	return vault.NewService(cnf.VaultAddr, cnf.VaultToken, cnf.DeploymentID)
+}
+
+func main() {
+	// Parse flags
+	filepath := flag.String("a", "outputs.yaml", "Outputs file path to save secrets from vault")
+	flag.Parse()
+
+	// Initialize and write to Vault stores for every component
+	initConfig()
+	secretStore := getVaultService("global")
+
+	// Get the list of App names from vault
+	apps, err := secretStore.ListAppNames()
+	if err != nil {
+		panic(err)
+	}
+
+	// Get the list of scopes by Splitting KAIGARA_SCOPES env
+	scopesList := strings.Split(cnf.Scopes, ",")
+	if len(scopesList) <= 0 {
+		panic("Wrong KAIGARA_SCOPES")
+	}
+	
+	// Create Secrets map
+	secretsMap := make(map[string]interface{})
+
+	// Create App map
+	appMap := make(map[string]interface{})
+
+	// Get the secrets from vault
+	for _, app := range apps {
+		
+		scopeMap := make(map[string]interface{})
+		scopeInit := make(map[string]interface{})
+		for _, scope := range scopesList {
+			err := secretStore.LoadSecrets(app, scope)
+			if err != nil {
+				panic(err)
+			}
+			secrets, err := secretStore.GetSecrets(app, scope)
+			if err != nil {
+				panic(err)
+			}
+
+			scopeMap[scope] = secrets
+		}
+		scopeInit["scopes"] = scopeMap
+		appMap[app] = scopeInit
+	}
+
+	secretsMap["secrets"] = appMap
+
+	// Marshal to yaml from map
+	res, err := yaml.Marshal(&secretsMap)
+	if err != nil {
+		panic(err)
+	}
+
+	// Dump secrets
+	fmt.Printf("--- secrets dump:\n%s\n\n", string(res))

```

Comment:
Could you please remove `--- secrets dump from here` so that we could send this output to a pipe when needed?

Reply:
Removed `--- secrets dump from here`